numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -309,6 +308,53 @@ static int
 
 /**end repeat**/
 
+static NPY_INLINE npy_longdouble
+string_to_long_double(PyObject*op)
+{
+    char *s;
+    char *end;
+    npy_longdouble temp;
+    PyObject* b;
+
+    if (PyUnicode_Check(op)) {
```

Comment:
Mostly I think it makes it easier to follow the code, otherwise six of one, half dozen of the other.